compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -698,7 +698,8 @@ private void loadErrorType(MethodVisitor mv, BErrorType errorType) {
             return;
         }
 
-        if (Symbols.isFlagOn(errorType.flags, Flags.ANONYMOUS)) {
+        boolean samePackage = JvmCodeGenUtil.isSameModule(this.packageID, pkgID);
+        if (Symbols.isFlagOn(errorType.flags, Flags.ANONYMOUS) && samePackage) {
```

Comment:
IMO correct fix would be to fix the hashcode and call get anon type. Shall we create an issue for that

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/39483.